packages/core/src/utils/summarizer.ts

```diff
@@ -0,0 +1,131 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ToolResult } from '../tools/tools.js';
+import {
+  Content,
+  GenerateContentConfig,
+  GenerateContentResponse,
+} from '@google/genai';
+import { GeminiClient } from '../core/client.js';
+import { DEFAULT_GEMINI_FLASH_MODEL } from '../config/models.js';
+import { PartListUnion } from '@google/genai';
+
+/**
+ * A function that summarizes the result of a tool execution.
+ *
+ * @param result The result of the tool execution.
+ * @returns The summary of the result.
+ */
+export type Summarizer = (
+  result: ToolResult,
+  geminiClient: GeminiClient,
+  abortSignal: AbortSignal,
+) => Promise<string>;
+
+/**
+ * The default summarizer for tool results.
+ *
+ * @param result The result of the tool execution.
+ * @param geminiClient The Gemini client to use for summarization.
+ * @param abortSignal The abort signal to use for summarization.
+ * @returns The summary of the result.
+ */
+export const defaultSummarizer: Summarizer = (
+  result: ToolResult,
+  _geminiClient: GeminiClient,
+  _abortSignal: AbortSignal,
+) => Promise.resolve(JSON.stringify(result.llmContent));
+
+// TODO: Move both these functions to utils
+function partToString(part: PartListUnion): string {
+  if (!part) {
+    return '';
+  }
+  if (typeof part === 'string') {
+    return part;
+  }
+  if (Array.isArray(part)) {
+    return part.map(partToString).join('');
+  }
+  if ('text' in part) {
+    return part.text ?? '';
+  }
+  return '';
+}
+
+function getResponseText(response: GenerateContentResponse): string | null {
```

Comment:
I'd be surprised if the genai SDK didn't have a way already to pull text from responses